example/ck_tile/35_batched_transpose/batched_transpose.cpp

```diff
@@ -0,0 +1,310 @@
+                    }
+                }
+                std::cout << "]" << std::endl;
+            }
+            std::cout << "]" << std::endl;
+        }
+        std::cout << std::endl;
+    }
+    std::cout << "--------------------" << std::endl;
+}
+
+// different threshold for different dtype
+template <typename DataType>
+auto get_elimit(std::string /*init_method*/)
+{
+    double rtol = 1e-3;
+    double atol = 1e-3;
+    return ck_tile::make_tuple(rtol, atol);
+}
+
+template <>
+auto get_elimit<ck_tile::bf16_t>(std::string /*init_method*/)
+{
+    double rtol = 1e-2;
+    double atol = 1e-2;
+    return ck_tile::make_tuple(rtol, atol);
+}
+
+template <>
+auto get_elimit<ck_tile::fp8_t>(std::string init_method)
+{
+    if(init_method == "ui" || init_method == "ni")
+    {
+        unsigned max_rounding_point_distance = 0;
+        double atol                          = 2e-3;
+        return ck_tile::make_tuple(max_rounding_point_distance, atol);
+    }
+    else
+    {
+        unsigned max_rounding_point_distance = 1;
+        double atol                          = 0.0625;
+        return ck_tile::make_tuple(max_rounding_point_distance, atol);
+    }
+}
+
+auto create_args(int argc, char* argv[])
+{
+    ck_tile::ArgParser arg_parser;
+    arg_parser.insert("v", "1", "weather do CPU validation or not")
+        .insert("pr", "fp16", "input data type. fp16/fp32 (representing 8/16/32 bit data)")
+        .insert("N", "1", "input batch size. ")
+        .insert("C", "15", "input channel size.")
+        .insert("H", "1", "input height size.")
+        .insert("W", "32", "input width size. ")
+        .insert("stride_dim0", "480", "input dim0 stride. ")
+        .insert("stride_dim1", "32", "input dim1 stride.")
+        .insert("stride_dim2", "32", "input dim2 stride.")
+        .insert("stride_dim3", "1", "input dim3 stride. ")
+        .insert("seed", "-1", "seed to be used, -1 means random every time")
+        .insert("kname", "0", "t to 1 will print kernel name");
+
+    bool result = arg_parser.parse(argc, argv);
+    return std::make_tuple(result, arg_parser);
+}
+
+template <typename Type>
+bool test_batched_transpose(ck_tile::ArgParser args)
+{
+    int validate     = args.get_int("v");
+    std::string prec = args.get_str("pr");
+    int N            = args.get_int("N");
+    int C            = args.get_int("C");
+    int H            = args.get_int("H");
+    int W            = args.get_int("W");
+    int stride_dim0  = args.get_int("stride_dim0");
+    int stride_dim1  = args.get_int("stride_dim1");
+    int stride_dim2  = args.get_int("stride_dim2");
+    int stride_dim3  = args.get_int("stride_dim3");
+    int seed         = args.get_int("seed");
+
+    std::string layout_in = "NCHW", layout_out = "NHWC";
+    int stride_out_dim0 = stride_dim0, stride_out_dim1 = W * C, stride_out_dim2 = C,
+        stride_out_dim3 = 1;
+    if(stride_dim0 == C * H * W && stride_dim1 == 1 && stride_dim2 == W * C && stride_dim3 == C)
```

Comment:
We should decide the stride configuration depending on the layout configuration right?